kubernetes/kube_provider.go

```
@@ -0,0 +1,98 @@
+
+// ToRESTConfig returns the configured REST config. Pointer is to the
+// struct on KubeProvider, so callers must not mutate it.
+func (p *KubeProvider) ToRESTConfig() (*restclient.Config, error) {
+	return &p.RestConfig, nil
+}
+
+// ToDiscoveryClient returns a disk-cached discovery client scoped to the
+// configured apiserver. The cache directory layout mirrors the kubectl
+// convention (~/.kube/cache/discovery/<hostname>) so concurrent kubectl
+// usage shares the cache.
+func (p *KubeProvider) ToDiscoveryClient() (discovery.CachedDiscoveryInterface, error) {
```

golangci-lint (2.12.2)

[error] SA4023(related information): (*github.com/alekc/terraform-provider-kubectl/kubernetes.KubeProvider).ToDiscoveryClient never returns a nil interface value

(staticcheck)